### DISCUSSION TITLE
Reload $PATH variable

### DISCUSSION THREAD

**Discussion by noximo** [gitextensions/gitextensions#10231] [Q&A] [answered]
Hi!

I'm developing several PHP applications, each can have a different PHP version. To switch between them I wrote a script that basically adds `PATH=/c/wamp64/bin/php/php8.1.0:$PATH` (with the proper version) to .bashrc. 

Then I need to simply run `bash` in CLI to update the path. 

Problem is that I cannot tell GitExtensions to run this command before running git commands. 
`git commit` then fails because the pre-commit hook calls composer and composer errors because of incompatible PHP versions. 

Is there a way to force GitExtensions to reload $PATH? Closing the app and running it again does nothing, updating it in Git Bash only updates that, not the GitExtensions. The only way right now is to restart the computer.

Any ideas?

**Comment by pmiossec** [comment, marked as answer]
$PATH of a process is inherited from the parent process that created it during the process creation.
So you can't refresh an env var of a process.
So you have to close the instance of GitExtensions and launch it again from a from having the PATH updated. Maybe in your case it is from the command line used to run the bash script.

>The only way right now is to restart the computer.

Killing all the 'explorer" process and relaunching one instance solve the issue... (Win+R or from process explorer)

**Reply by noximo** [reply]
Thank you! 

Restarting explorer.exe did the trick.
